src/Microsoft.Diagnostics.Monitoring.WebApi/CollectionRulePipelineStateHolder.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System;
+
+namespace Microsoft.Diagnostics.Monitoring.WebApi
+{
+    internal record class CollectionRulePipelineStateHolder
```

Comment:
I think this should be combined with `CollectionRuleStateHolder`. I don't think there is a need to keep them separate.